librustls/src/client.rs

```diff
@@ -784,6 +787,74 @@ impl rustls_client_config {
             rustls_result::Ok
         }
     }
+
+    /// Create a new client `rustls_connection` with custom ALPN protocols.
```

Comment:
Did you consider having `_new()` call `_new_alpn()` like we do in Rust-land? Seems nice to avoid the duplication?

Reply:
I was avoiding that initially to not require an extra `clone()` for the `_new()` path, but since that will happen internal to rustls anyway in retrospect it's better to avoid the duplication. I'll adjust.

Reply:
Reworked so both versions share a common impl.